File: packages/app/src-capacitor/ios/App/App/Plugins/AppVersion/AppNativeInfoPlugin.m
//
//  AppVersionPlugin.m
//  App
//
//  Created by gjm on 2023/4/19.
//

#import <Capacitor/Capacitor.h>

CAP_PLUGIN(AppNativeInfoPlugin, "AppNativeInfo",
    CAP_PLUGIN_METHOD(getInfo, CAPPluginReturnPromise);
)

File: packages/app/src-capacitor/ios/App/App/Plugins/AppVersion/AppNativeInfoPlugin.swift
//
//  AppVersionPlugin.swift
//  App
//
//  Created by gjm on 2023/4/18.
//

import Capacitor
import VaultBaseFramework

@objc(AppNativeInfoPlugin)
public class AppNativeInfoPlugin: CAPPlugin {
    @objc func getInfo(_ call: CAPPluginCall) {
          
        let infoDictionary = Bundle.main.infoDictionary
        
        guard let majorVersion = infoDictionary?["CFBundleShortVersionString"] else {
            call.reject("get app version error")
            return
        }
        
        guard let appVersionCode = infoDictionary?["CFBundleVersion"] else {
            call.reject("get minor version error")
            return
        }
        
        call.resolve(["appVersionName": majorVersion,"appVersionCode":appVersionCode])
    }
}
